ui/src/app/Router.jsx

```diff
@@ -60,11 +61,11 @@ class Router extends Component {
   fetchIfNeeded() {
     const { metadata, messages } = this.props;
 
-    if (metadata.shouldLoad) {
+    if (metadata.shouldLoad && !metadata.isError) {
       this.props.fetchMetadata();
     }
 
-    if (messages.shouldLoad) {
+    if (messages.shouldLoad && !messages.isError) {
       this.fetchMessages();
     }
```

Comment:
Noob question, but would it be possible to merge these two if clauses? They look identical to me.

Reply:
I misread that, sorry. 😞

Reply:
No problem, the two terms are indeed very similar :)